### PR TITLE
[bitfinex 1] added cancelAllOrders and replaceOrder methods

### PR DESCRIPTION
**Pull request by npomfret**

### CODE REVIEW

File: js/bitfinex.js [hunk 1]
```diff
@@ -653,6 +653,19 @@ module.exports = class bitfinex extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        let order = this.prepareOrder (symbol, type, side, amount, price);
+        let result = await this.privatePostOrderNew (this.extend (order, params));
+        return this.parseOrder (result);
+    }
+
+    async replaceOrder (id, symbol, type, side, amount, price = undefined, params = {}) {
```
Comment (kroitor) [inline js/bitfinex.js]: We have `editOrder` instead, can you please rename it? )

File: js/bitfinex.js [hunk 2]
```diff
@@ -658,7 +658,7 @@ module.exports = class bitfinex extends Exchange {
         return this.parseOrder (result);
     }
 
-    async replaceOrder (id, symbol, type, side, amount, price = undefined, params = {}) {
+    async editOrder (id, symbol, type, side, amount, price = undefined, params = {}) {
```
Comment (kroitor) [inline js/bitfinex.js]: There's still one more issue with the amount – it isn't required in most cases of editOrder. Therefore we have to properly account for that when we prepare the request. In other words, I wouldn't unify `prepareRequest` for `createOrder` and `editOrder` and would rather keep them separate, or it has to be edited further to account for an undefined amount.

**Comment by kroitor** [comment]
> Can you explain? It is mandatory for the bitfinex api call to work

Basically, we only check if the argument is defined or not, if it has a default value of `undefined` in the method signature. Because the `id` doesn't have a default value of `undefined` – we don't have to check for that.

In fact, very few users will actually try to edit an order without an id.

**Comment by npomfret** [comment]
@kroitor FYI (i don't know if it matters) on bitfinex this method is a cancel and replace, rather than an _edit_  so you get a new order id.  The method name is a bit misleading

**Comment by kroitor** [comment]
@npomfret 

> FYI (i don't know if it matters) on bitfinex this method is a cancel and replace, rather than an edit so you get a new order id. The method name is a bit misleading

Yeah, but in fact, this is how most of exchanges behave – they will return a new order with a new id in it, which is then returned to the user from editOrder, so, if the user switches to the new id for that order (he should, anyway), this won't be a problem, right?

**Comment by npomfret** [comment]
Yeah, as long as the user knows switch id should it change they won't have any problem.

I know that liquid for example keeps your id the same.  It won't let you change a partially filled order for this reason, in which case you need to cancel and place a new one yourself.

**Pull request merged**